src/test/make_app_test.ts

```diff
@@ -47,12 +48,82 @@ suite('makeApp', () => {
 
   test('shows friendly error when bower.json does not exist', () => {
     let called = false;
-    console.error = function(e) {
+    console.error = function(e: any) {
       called = true;
     };
     const app = makeApp({root: path.resolve(__dirname, 'no_bower_json/')});
     assert.isFalse(called);
     assert.equal(app.packageName, 'no_bower_json');
   });
 
+  suite('compilation', () => {
+
+    const testCompilation = ({url, agent, compile, result}: {
+      url: string,
+      agent?: string,
+      compile: 'always' | 'never' | 'auto',
+      result: 'compiled' | 'uncompiled'
+    }) => async() => {
+      const app = makeApp({
+        root: __dirname,
+        componentDir: path.join(root, 'bower_components'), compile,
+      });
+      let tester = supertest(app).get(url);
+      if (agent) {
+        tester = tester.set('User-Agent', agent)
+      }
+      await tester.expect((res: Response) => {
```

Comment:
that's... odd. `supertest.Test` is a `Test`, `Request`, _and_ a `Promise<Response>`? 🙄